include/deal.II/base/parameter_acceptor.h

```diff
@@ -516,6 +520,72 @@ class ParameterAcceptor : public Subscriptor
 
 
 
+/**
+ * A proxy ParameterAcceptor wrapper for classes that have a static member
+ * function @p declare_parameters, and a non virtual @p parse_parameters method.
+ *
+ * If a class already implements the static member function declare_parameters,
+ * and the method declare_parameters, then it cannot be derived directly from
+ * ParameterAcceptor, as the existing member functions would be incompatible
+ * with the virtual methods with the same name that ParameterAcceptor expects
+ * you to overload. For these cases, you can still use ParameterAcceptor
+ * facilities and logic by wrapping your class inside this proxy. The resulting
+ * class will have the public interface of your @p SourceClass, and will provide
+ * the correct interface to ParameterAcceptor.
+ *
+ * An example usage is given by the following snippet of code, using
+ * Functions::ParsedFunction as an example source class:
+ *
+ * @code
+ * ParameterAcceptorProxy<Functions::ParsedFunction<2> > fun("Some function");
+ * ParameterAcceptor::initialize("test.prm");
+ * @endcode
+ *
+ * The above snippet of code will initialize ParameterAcceptor::prm with a
+ * section "Some function", and will correctly parse and assing to the object
+ * `fun` the expression parsed from the file `test.prm`. If non existant, the
+ * program will exit, and generate it for you (here you can see the resulting
+ * short text version of the parameter file generated with the above snippet):
+ *
+ * @code
+ * # Parameter file generated with
+ * # DEAL_II_PACKAGE_VERSION = 9.0.0-pre
+ * subsection Some function
+ *   set Function constants  =
+ *   set Function expression = 0
+ *   set Variable names      = x,y,t
+ * end
+ * @endcode
+ *
+ * @author Luca Heltai, 2018
+ */
+template<class SourceClass>
+class ParameterAcceptorProxy : public SourceClass, public ParameterAcceptor
+{
+public:
+  /**
+   * Default constructor. The argument `section_name` is forwarded to the
+   * constructor of the ParameterAcceptor class, while all other arguments
+   * are passed to the SourceClass constructor.
+   */
+  template<typename... Args>
+  ParameterAcceptorProxy(const std::string section_name, Args... args);
+
+  /**
+   * Overloads the ParameterAcceptor::declare_parameters function. See the base
+   * class for further information.
```

Comment:
for this function and the next, I suppose that it simply calls the corresponding function in `SourceClass`?